### PR TITLE
Range slider. Phase 0

### PR DESCRIPTION
**Pull request by Ta4i**
Implementation of Range Slider according to https://github.com/wix/stylable-components/tree/master/docs/range-slider

### CODE REVIEW

**Review by tyv — CHANGES_REQUESTED** [review]

File: demo/components/slider-demo.tsx [hunk 1]
```diff
@@ -15,8 +17,10 @@ export class SliderDemo extends React.Component<{}, SliderDemoState> {
         super(props);
 
         this.state = {
-            value: 50,
-            rawValue: '50'
+            value: [50],
```
[the remark this answers is not in the document]
Comment (Ta4i) [reply demo/components/slider-demo.tsx]: Value signature changed for Slider. Both Simple and Range Sliders have array type of value.

File: src/utils/nearest.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+            if (item === nearestValue) {
+                indexes.push(index);
+            }
+            return indexes;
+        }, []);
+        return  value < nearestValue ? sameValueIndexes[0] : last(sameValueIndexes);
+    } else {
+        return vector.indexOf(nearestValue);
+    }
+}
```
[the remark this answers is not in the document]
Comment (Ta4i) [reply src/utils/nearest.ts]: Sure. Done

File: demo/components/slider-demo.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (tyv) [reply demo/components/slider-demo.tsx]: but it is inconvenient

[the remark this answers is not in the document]
Comment (Ta4i) [reply demo/components/slider-demo.tsx]: Ok, agree. Most frequent case is Simple Slider and it would look strange to use array as value type for it.

**Pull request merged**